test/gas/gas.js

```diff
@@ -0,0 +1,563 @@
+const assert = require("assert");
+const sleep = require("../helpers/utils/sleep");
+const bootstrap = require("../helpers/contract/bootstrap");
+const getWeb3 = require("../helpers/web3/getWeb3");
+const isGasExpenseCorrect = require("./lib/isGasExpenseCorrect");
+const testTransactionEstimate = require("./lib/transactionEstimate");
+const toBytes = require("./lib/toBytes");
+const deployContract = require("./lib/customContractDeploy");
+
+const RSCLEAR_REFUND = 15000;
+const RSCLEAR_REFUND_FOR_RESETTING_DIRTY_SLOT_TO_ZERO = 19800;
+const RSELFDESTRUCT_REFUND = 24000;
+
+// Thanks solc. At least this works!
+// This removes solc's overzealous uncaughtException event handler.
+process.removeAllListeners("uncaughtException");
+
+describe("Gas", () => {
+  const mainContract = "EstimateGas";
+  const contractFilenames = [];
+  const contractPath = "../../contracts/gas/";
+
+  const mnemonic = "candy maple cake sugar pudding cream honey rich smooth crumble sweet treat";
+  const options = { mnemonic };
+  const services = bootstrap(mainContract, contractFilenames, options, contractPath);
+
+  let deploymentReceipt;
+
+  before("Adding funds to the contract", async function() {
+    this.timeout(10000);
+    const { accounts, bytecode, contract } = services;
+
+    const promiEvent = contract.deploy({ data: bytecode }).send({
+      from: accounts[0],
+      gas: 3141592
+    });
+
+    promiEvent.on("receipt", (receipt) => {
+      deploymentReceipt = receipt;
+    });
+  });
+
+  describe("Refunds", () => {
+    it(
+      "accounts for Rsclear Refund in gasEstimate when a dirty storage slot is reset and it's original " +
+        " value is 0",
+      async() => {
```

Comment:
we shouldn't be using arrow functions to encapsulate tests

https://remarkablemark.org/blog/2016/08/20/arrow-functions-in-mocha/

Reply:
@ben, as far as arrow functions in tests, the _only_ reason not to is if you really _want_ to use `this` inside of the test to reference the test instance itself. There are no instances where we actually need to do that within an `it`. Within `before` we absolutely _can't_ use arrow functions.

I like the arrow functions here, personally. :-)